tap/xdp/ipproto.c

```diff
@@ -0,0 +1,79 @@
+// +build ignore
+
+#include <linux/bpf.h>
+#include <linux/in.h>
+#include <linux/udp.h>
+#include <linux/if_ether.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+
+
+#define MAX_SOCKS 64
+
+static volatile unsigned const char PROTO;
+static volatile unsigned const char PROTO = IPPROTO_ICMP;
+
+//Ensure map references are available.
+/*
+        These will be initiated from go and
+        referenced in the end BPF opcodes by file descriptor
+*/
+
+struct bpf_map_def SEC("maps") xsks_map = {
+	.type = BPF_MAP_TYPE_XSKMAP,
+	.key_size = sizeof(int),
+	.value_size = sizeof(int),
+	.max_entries = MAX_SOCKS,
+};
+
+struct bpf_map_def SEC("maps") qidconf_map = {
+	.type = BPF_MAP_TYPE_ARRAY,
+	.key_size = sizeof(int),
+	.value_size = sizeof(int),
+	.max_entries = MAX_SOCKS,
+};
```

Comment:
Use the macros defined in `maps.h`.

Reply:
Fixed with https://github.com/up9inc/mizu/commit/b4a9be6ad3bdf514ee3bc265fb15b921196730da